### PR TITLE
Allow configuring Default Volumesnapshot and Storage class

### PR DESCRIPTION
**Pull request by rakesh-garimella**
**How to categorize this PR?**

/kind enhancement
/platform azure

**What this PR does / why we need it**:

+ create a default VolumeSnapshotClass for Azure.
+ allow configuring default annotation for SC.
+ allow configuring default annotation for VSClass.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:
```breaking user
Provider-azure now deploys a new default `volumesnapshotclass`. 
```
```feature user
You can now configure if the deployed `storageclass` and `volumesnapshotclass` are marked as default.
```

### CODE REVIEW

**Comment by gardener-robot-ci-1** [comment]
Thank you @rakesh-garimella for your contribution. Before I can start building your PR, a member of the organization must set the required label(s) {'reviewed/ok-to-test'}. Once started, you can check the build status in the PR checks section below.

File: charts/internal/shoot-storageclasses/templates/storageclasses.yaml [hunk 1]
```diff
@@ -68,3 +68,12 @@ parameters:
 volumeBindingMode: WaitForFirstConsumer
 allowVolumeExpansion: true
 {{- end }}
+---
+apiVersion: snapshot.storage.k8s.io/v1beta1
+kind: VolumeSnapshotClass
+metadata:
+  name: default
+  annotations:
+    snapshot.storage.kubernetes.io/is-default-class: “true”
```
Comment (rfranzke) [inline charts/internal/shoot-storageclasses/templates/storageclasses.yaml:77]:
```suggestion
    snapshot.storage.kubernetes.io/is-default-class: "true"
```

**Comment by kon-angelo** [comment]
@rakesh-garimella I added to the PR a mechanism to enable or disable the default annotations so that it won't collide with existing ones. We can proceed with it now.

**Review by MartinWeindel — APPROVED** [review]
/lgtm

**Pull request merged**